Peer Review for Linux

Review and test this guide by a peer member with the following prespectives
 -  by Firefox
 
 README
 ---
 - [ ] Ensure that command substitutions use the `$(cmd)` format and not ``` `cmd` ```
 - [ ] Ensure that the console output of the integrated tests do not include the `[INFO]` tab
 - [ ] Check the quality and presentation of guide according to [Structure and Style Guideline](https://github.com/OpenLiberty/draft-guides-template/wiki/Guidelines-for-Structure-and-Styles)
 - [ ] Check the appearance of the guide:
    - [ ] Table of contents
    - [ ] Headings
    - [ ] Paragraphs
    - [ ] code snippets
    - [ ] outputs
    - [ ] links (shoud open a new tab)
    - [ ] hotspots (should highlight the code correctly)
 - [ ] Check that the `pom.xml`, `server.xml`, and java files etc files are clean on the right code pane
    - [ ] Remove any dependencies and features that are not required
    - [ ] Ensure that the parent pom.xml files do not contain any dependencies, and all dependencies are specified in the children pom.xml files
    - [ ] lines are not too long and wrapped
    - [ ] lines are aligned nicely
 - [ ] Examples with the right outcomes are provided
 - [ ] Examples with the wrong outcomes, if any, are provided
 - [x] Run Acrolinx Checker (above 70 score approximately)
 
 `pom.xml`
 ---
 - [ ] Check that the file follows the template found here: https://github.com/OpenLiberty/draft-guides-template/blob/draft/finish/pom.xml
 - [ ] Check that the dependencies and plugins are latest in all of the `pom.xml` files:
    - jakarta.jakartaee-api
    - microprofile
    - liberty-maven-plugin
    - etc.
 - [ ] Check that there are 4 spaces per indent for proper formatting
 - [ ] Check that no tab characters are used
 
 Overall checks:
 ---
 - [ ] Check the consistency of guide with the template and other guides
 - [ ] Check the quality of code according to the best coding practices
 - [ ] Check that all licensing statements are properly stated in all files, with the correct year (Should be present in all Java files + the `index.html`)
 - [ ] Check that the directories are properly structured
 - [ ] Check that some of these `page-tags` are used in a guide: `MicroProfile, Maven, Docker, Kubernetes, Gradle, Jakarta EE, Security, Cloud`. Only these tags are visible on the website. Latest list [here](https://github.com/OpenLiberty/guides-common/blob/master/guide_tags.json).
 - [ ] Check the `attribution` statement is accurate for the guide
 - [ ] Verify the GitHub Actions scripts, if any, are accurate and consistent with other guides
 
 Additional tests:
 ---
 - [ ] tests where applicable:
    - [ ] Define test coverage and review with team (including guide contributor, if available)
    - [ ] Define detail test cases
    - [ ] Consider corner cases targeting the specific guide
    - [ ] Consider corner cases UI tests
    - [ ] Consider testing the `curl` command for localhost URLs
    - [ ] Make sure the `index.html` information is correct by visiting the root, e.g. `http://localhost:9080`
 - [ ] Testing with different IDEs, ie, Atom, Eclipse (Optional: VS.code, IntelliJ, Microclimate)
    - import the finish directory and see any complication error, unused import packages, etc
 - [ ] Consider SEO title and description for the guides
 - [ ] Ensure automated tests with GitHub Actions are enabled
 - [ ] Run `diff -r start/ finish/` and there is no differences expect `tag` and `end` comments 
 - [ ] Ensure that the automation tests are able to run when PR is created
 - [ ] Test the guide end-to-end with working instruction and sample code with either of the following Java
    - [ ] Java 11
    - [ ] Java 17
 - [ ] Perform all the defined test cases

comments from @Ruilin-Ma

- Documentation link return 404
  - You can now develop the application and the different modules together in dev mode by using the Liberty Gradle plug-in. To learn more about how to use dev mode with multiple modules, check out the [Documentation](https://github.com/OpenLiberty/ci.gradle/blob/main/docs/dev.md#multiple-modules). ==> documentation 404
- While running `./gradlew <child project>:build` in start directory (Building the multi-module application section), there were two java.net.ConnectException were thrown while packaging Java file:
```
root@c26926v1:~/test/draft-guide-gradle-multimodules/start# ./gradlew ear:build

> Task :ear:test

IT > testHeightsPage() FAILED
    java.net.ConnectException at IT.java:46

IT > testIndexPage() FAILED
    java.net.ConnectException at IT.java:22

2 tests completed, 2 failed

> Task :ear:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ear:test'.
> There were failing tests. See the report at: file:///root/test/draft-guide-gradle-multimodules/start/ear/build/reports/tests/test/index.html

* Try:
> Run with --scan to get full insights.

BUILD FAILED in 4s
7 actionable tasks: 1 executed, 6 up-to-date
```

--- a/README.adoc
+++ b/README.adoc
@@ -218,7 +218,7 @@ Set the [hotspot=allprojects file=1]`basic configuration` for the project. Apply
 
 == Developing the application
 
-You can now develop the application and the different modules together in dev mode by using the Liberty Gradle plug-in. To learn more about how to use dev mode with multiple modules, check out the https://github.com/OpenLiberty/ci.gradle/blob/main/docs/dev.md#multiple-modules[Documentation^].
+You can now develop the application and the different modules together in dev mode by using the Liberty Gradle plug-in. To learn more about how to use dev mode with multiple modules, check out the https://github.com/OpenLiberty/ci.gradle/blob/main/docs/libertyDev.md#multiple-modules[Documentation^].
 
 // cloud hosted instructions
 ifdef::cloud-hosted[]

--- a/README.adoc
+++ b/README.adoc
@@ -218,7 +218,7 @@ Set the [hotspot=allprojects file=1]`basic configuration` for the project. Apply
 
 == Developing the application
 
-You can now develop the application and the different modules together in dev mode by using the Liberty Gradle plug-in. To learn more about how to use dev mode with multiple modules, check out the https://github.com/OpenLiberty/ci.gradle/blob/main/docs/libertyDev.md#multiple-modules[Documentation^].
+You can now develop the application and the different modules together in dev mode by using the Liberty Gradle plug-in. To learn more about how to use dev mode with multiple modules, check out the https://github.com/OpenLiberty/ci.gradle/blob/main/docs/libertyDev.md#multi-project-builds[Documentation^].
 
 // cloud hosted instructions
 ifdef::cloud-hosted[]

--- a/finish/ear/build.gradle
+++ b/finish/ear/build.gradle
@@ -63,6 +63,7 @@ liberty {
 // tag::test[]
 test {
     systemProperty 'http.port', liberty.server.var.'http.port'
+    enabled = gradle.startParameter.taskNames.contains('test')
 }
 // end::test[]
 

